backend/app/Http/Controllers/ProfileController.php

```diff
@@ -65,19 +66,45 @@ public function updateMe()
 
     /**
      * Gets all the cards for a given user. Includes hidden cards if the authorized user is requesting their own profile.
+     * Includes isFollowing, true if authorized user is following user_id
      *
      * @return mixed cards
      */
     public function getUserDetail($user_id)
     {
-        $isRequestingMe = auth()->user() && (auth()->user()->id == $user_id);
-
+        $user = auth()->user();
+        $isRequestingMe = $user && ($user->id == $user_id);
         $cards = Card::with('attributes');
+        $isFollowing = false;
         if (!$isRequestingMe) {
             $cards = $cards->where(Card::FIELD_HIDDEN_TOGGLE, false);
+            $isFollowing = $user->isFollowing($user_id);
         }
         $cards = $cards->where('user_id', $user_id)->get();
 
-        return response()->build(self::RESPONSE_MESSAGE_SUCCESS, ['cards'=> $cards]);
+        return response()->build(self::RESPONSE_MESSAGE_SUCCESS, ['cards'=> $cards, 'isFollowing'=> $isFollowing]);
+    }
+
+    /**
+     * authorized user follows user_id
+     * 
+     * @return const RESPONSE_MESSAGE_SUCCESS or RESPONSE_MESSAGE_ALREADY_FOLLOWING
+     */
+    public function follow($user_id)
+    {
+        $user = auth()->user();
+
+        $response = '';
```

Comment:
it could make sense for this logic to be inside the model, and pass $user-> and $user_id, and then have it return true or false 🤷‍♂️